runtime/message.cpp

```diff
@@ -396,14 +404,18 @@ size_t MessageBase::encode(ostream& to) const
 }
 
 //-------------------------------------------------------------------------------------------------
-size_t MessageBase::encode_group(const unsigned short fnum, char *to) const
+size_t MessageBase::encode_group(const unsigned short fnum, char *to, const char *endOfSafeEncode) const
 {
 	const char *where(to);
 	GroupBase *grpbase(find_group(fnum));
 	if (!grpbase)
 		throw InvalidRepeatingGroup(fnum, FILE_LINE);
-	for (const auto *pp : grpbase->_msgs)
-		to += pp->encode(to);
+	for (const auto *pp : grpbase->_msgs) {
+		if (to  > endOfSafeEncode) {
+		  throw EncodedMessageExceededBuffer(FIX8_MAX_MSG_LENGTH + HEADER_CALC_OFFSET);
+		}
```

Comment:
Same as above - I think this check should be after next line.
Additionally, it may be more readable to pass something like `endOfSafeEncode - where` to the exception.